ci: re-enable RBE for http tests

alan-agius4 opened the pull request:
This commit enables RBE for common http tests which is now possible as the latest version of build-tooling supports Node.js 18.

alan-agius4 commented on package.json:
```diff
@@ -46,11 +46,11 @@
-    "@angular-devkit/build-angular": "16.0.0-rc.4",
-    "@angular-devkit/core": "16.0.0-rc.4",
-    "@angular-devkit/schematics": "16.0.0-rc.4",
+    "@angular-devkit/build-angular": "16.1.0",
```
These updates are needed as it seems that the direct dependency on `@angular-devkit/build-angular` from `@angular/build-tooling` is not being used due to:

Due to 
```
js_library(
    name = "angular_devkit_plugins",
    package_name = "@angular-devkit/build-angular/src/babel/plugins",
    srcs = [
        "@npm//:node_modules/@angular-devkit/build-angular/src/tools/babel/plugins/adjust-static-class-members.js",
        "@npm//:node_modules/@angular-devkit/build-angular/src/tools/babel/plugins/adjust-typescript-enums.js",
        "@npm//:node_modules/@angular-devkit/build-angular/src/tools/babel/plugins/elide-angular-metadata.js",
        "@npm//:node_modules/@angular-devkit/build-angular/src/tools/babel/plugins/pure-toplevel-functions.js",
    ],
    deps = [
        "@npm//@babel/core",
        "@npm//@babel/helper-annotate-as-pure",
    ],
)
```

devversion replied:
Yes, Bazel deps always go through the dependencies of framework. So this makes sense

alan-agius4 replied:
In that case, to reduce the dependency tree, does it make sense to remove @angular-devkit/build-angular for the deps of build-tooling?

devversion replied:
Yes, we can do that. I don't fully recall why we never did it. I think initially I wanted to have these as peer dependencies. Should be easier to handle with `rules_js` eventually

alan-agius4 committed:
ci: re-enable RBE for http tests

This commit enables RBE for common http tests which is now possible as the latest version of build-tooling supports Node.js 18.

devversion reviewed (approved):
LGTM

Reviewed-for: global-dev-infra-approvers

AndrewKushnir commented:
This PR was merged into the repository by commit 232a78630eeeead1e033be65e0ce115f6ec4d0b9.
